cou/steps/plan.py

```diff
@@ -55,14 +58,48 @@
 logger = logging.getLogger(__name__)
 
 
+class PlanWarnings:
+    """Representation of a collection of warning messages from plan generation.
+
+    This is a singleton class which holds all warning messages returned by applications
+    when generating a plan.
+    """
+
+    _instance = None
+    _warnings: list[str] = []
+
+    def __new__(cls) -> PlanWarnings:
+        """Create a new instance of the PlanWarnings class if doesn't already exist."""
+        if not cls._instance:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def add_warning(self, warning: str) -> None:
+        """Add a new warning to the collection.
+
+        :param warning: A warning message to be stored.
+        :type warning: str
+        """
+        self._warnings.append(warning)
+
+    @property
+    def warnings(self) -> list[str]:
+        """Get all warnings stored in the class.
+
+        :return: The list of all warnings stored in the class.
+        :rtype: list[str]
+        """
+        return self._warnings
+
+
```

Comment:
```suggestion
class PlanWarnings:
    """Representation of a collection of warning messages from plan generation.

    This is a singleton class which holds all warning messages returned by applications
    when generating a plan.
    """

    messages: list[str] = []

    @classmethod
    def add_message(cls, message: str) -> None:
        """Add a new warning to the collection.

        :param message: A warning message to be stored.
        :type message: str
        """
        cls.messages.append(message)
```

Reply:
We can remove `_instance` and basically whole `__new__`, since we are defining just list of strings, and I think we could go with classmethod so it's more easy to access those objects.

e.g.
```python
PlanWarnings.add_message("my message")
PlanWarnings.messages
# instead of
PlanWarnings().add_message("my message")
PlanWarnings().messages
```

Reply:
Done. Thanks for the suggestions